### PR TITLE
build(deps): update to RN 0.69.10

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [MOPLAT-720] 

### Description

Most changes come from [this guide](https://react-native-community.github.io/upgrade-helper/?from=0.67.5&to=0.69.10)

The diff might look big due to some breaking changes, now we can opt-in to the new architecture if we want and there is some flag logic about that on the native side.

Some things to note:
- added `[deprecated-react-native-prop-types](http://npmjs.com/package/deprecated-react-native-prop-types)` due to react-native removing some prop types.
- had to patch some of our deps due to this (see hacks.md and /patches):
    |patch|reason|
    |---|---|
    | patches/segment+analytics-react-native-appboy+1.5.0.patch | missmatch on kotlin version |
    |patches/react-native+0.69.10.patch|removal of prop types|
    |patches/react-native-credit-card-input+0.4.1.patch|removal of prop types|
    |patches/react-native-scrollable-tab-view+1.0.0.patch|removal of prop types|
- fixed some failing tests (nothing to worry about there)
- bonus update -> `React is version 18.0.0` 🤯
- RN -> 0.69.10
- hermes comes prebundled from this version onwards 🔥 so it will be easier to integrate/enable

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- update to react-native 0.69.10

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-720]: https://artsyproduct.atlassian.net/browse/MOPLAT-720?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/app/src/main/jni/MainApplicationModuleProvider.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  //
+  // auto module = samplelibrary_ModuleProvider(moduleName, params);
+  // if (module != nullptr) {
+  //    return module;
+  // }
+  // return rncore_ModuleProvider(moduleName, params);
+  return rncore_ModuleProvider(moduleName, params);
+}
+} // namespace react
+} // namespace facebook
```
Comment (gkartalis) [inline android/app/src/main/jni/MainApplicationModuleProvider.cpp:20]:
```suggestion
} // namespace facebook

```

File: android/gradle.properties [hunk 2]
```diff
@@ -25,4 +25,15 @@ android.useAndroidX=true
...
+# Use this property to specify which architecture you want to build.
+# You can also override it from the CLI using
+# ./gradlew <task> -PreactNativeArchitectures=x86_64
+reactNativeArchitectures=armeabi-v7a,arm64-v8a,x86,x86_64
+# Use this property to enable support to the new architecture.
+# This will allow you to use TurboModules and the Fabric render in
+# your application. You should enable this flag either if you want
+# to write custom TurboModules/Fabric components OR use libraries that
+# are providing them.
+newArchEnabled=false
```
Comment (gkartalis) [inline android/gradle.properties:39]:
```suggestion
newArchEnabled=false

```

File: package.json [hunk 3]
```diff
@@ -273,7 +274,7 @@
     "danger": "11.2.3",
     "dedent": "0.7.0",
     "dotenv": "8.2.0",
-    "eslint": "8.32.0",
+    "eslint": "7.32.0",
```
Comment (gkartalis) [inline package.json:277]:
```suggestion
    "eslint": "8.32.0",
```

Comment (gkartalis) [reply package.json:277]: seems like react-native is behind us on eslint, reverting to use our current version

File: android/settings.gradle [hunk 4]
```diff
@@ -1,3 +1,10 @@
 rootProject.name = 'Artsy'
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
+includeBuild('../node_modules/react-native-gradle-plugin')
+if (settings.hasProperty("newArchEnabled") && settings.newArchEnabled == "true") {
+    include(":ReactAndroid")
+    project(":ReactAndroid").projectDir = file('../node_modules/react-native/ReactAndroid')
+    include(":ReactAndroid:hermes-engine")
+    project(":ReactAndroid:hermes-engine").projectDir = file('../node_modules/react-native/ReactAndroid/hermes-engine')
+}
```
Comment (gkartalis) [inline android/settings.gradle:10]:
```suggestion
}

```

File: android/app/build.gradle [hunk 5]
```diff
@@ -268,3 +349,11 @@ task copyDownloadableDepsToLibs(type: Copy) {
...
 
 apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
+
+def isNewArchitectureEnabled() {
+    // To opt-in for the New Architecture, you can either:
+    // - Set `newArchEnabled` to true inside the `gradle.properties` file
+    // - Invoke gradle with `-newArchEnabled=true`
+    // - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
+    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+}
```
Comment (gkartalis) [inline android/app/build.gradle:359]:
```suggestion
}

```

File: Gemfile [hunk 6]
```diff
@@ -2,7 +2,7 @@ source 'https://rubygems.org'
 
 gem 'fastlane'
 
-gem 'cocoapods', '1.11.3'
+gem 'cocoapods', '>= 1.11.2'
```
Comment (gkartalis) [inline Gemfile:5]: @brainbicycle that was coming from the rn diff gemfile from the upgrade site, do you think we should change it to:

```suggestion
gem 'cocoapods', '>= 1.11.3'
```

Comment (brainbicycle) [reply Gemfile:5]: I think it is probably fine 🤷

File: ios/Podfile.lock [hunk 7]
```diff
@@ -104,64 +104,65 @@ PODS:
     - GoogleUtilities/Environment (~> 7.7)
     - GoogleUtilities/Logger (~> 7.7)
     - nanopb (~> 2.30908.0)
-  - Flipper (0.189.0):
+  - Flipper (0.125.0):
```
Comment (brainbicycle) [inline ios/Podfile.lock:107]: I see the migration doc removes the flipper line from the podfile but it looks like it resulted in flipper getting downgraded?

Comment (brainbicycle) [reply ios/Podfile.lock:107]: aha, you can specify version in Podfile and gradle properties: https://fbflipper.com/docs/getting-started/react-native/

**Review by brainbicycle — APPROVED** [review]
🙌 wooo! 🚀

**Review by damassi — APPROVED** [review]
Amazing, now we're in sync with Energy 🎉

**Comment by damassi** [comment]
@gkartalis - mind updating the PR description with a list of the patches added in this PR?

**Pull request merged**